packages/react-components/react-link/library/src/components/Link/useLinkState.ts

```diff
@@ -35,14 +36,24 @@ export const useLinkState_unstable = (state: LinkState): LinkState => {
   };
 
   // Disallow keydown event when component is disabled and eat events when disabledFocusable is set to true.
-  state.root.onKeyDown = (ev: React.KeyboardEvent<HTMLAnchorElement & HTMLButtonElement>) => {
-    if ((disabled || disabledFocusable) && (ev.key === Enter || ev.key === Space)) {
-      ev.preventDefault();
-      ev.stopPropagation();
-    } else {
-      onKeyDown?.(ev);
-    }
-  };
+  state.root.onKeyDown = mergeCallbacks(
+    state.root.onKeyDown,
```

Comment:
wont this break current behaviour when disabled ? this merging callback will invoke provided onKeyDown then the custom function ( only the custom function contains disabling resolution behaviours )